test/type_checker_test.dart

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:collection';
+
+import 'package:analyzer/dart/element/type.dart';
+import 'package:build_test/build_test.dart';
+import 'package:meta/meta.dart';
+import 'package:source_gen/source_gen.dart';
+import 'package:test/test.dart';
+
+void main() {
+  // Resolved top-level types from dart:core and dart:collection.
+  DartType staticMap;
+  DartType staticHashMap;
+  TypeChecker staticMapChecker;
+  TypeChecker staticHashMapChecker;
+
+  setUpAll(() async {
+    final resolver = await resolveSource('');
+
+    final core = resolver.getLibraryByName('dart.core');
+    staticMap = core.getType('Map').type;
+    staticMapChecker = new TypeChecker.fromStatic(staticMap);
```

Comment:
Might as well just create this inline in the test that uses them below like the others instead of creating them for all tests (same for `staticHashMapChecker`).